frontends/web/src/components/aopp/aopp.tsx

```diff
@@ -38,164 +38,127 @@ const Banner = ({ children }: TProps) => (
   <div className={styles.banner}>{children}</div>
 );
 
-type State = {
-  accountCode: accountAPI.AccountCode;
-  aopp?: aoppAPI.Aopp;
-}
-
-type Props = TranslateProps;
+const setAccountCodeDefault = (aopp: aoppAPI.Aopp | undefined, setAccountCode: React.Dispatch<React.SetStateAction<string>>) => {
+  if (aopp === undefined || aopp.state !== 'choosing-account') {
+    return;
+  }
+  if (aopp.accounts.length) {
+    setAccountCode(aopp.accounts[0].code);
+  }
+};
 
 const domain = (callback: string): string => new URL(callback).host;
 
-class Aopp extends Component<Props, State> {
-  public readonly state: State = {
-    accountCode: '',
-    aopp: undefined,
-  };
-  private unsubscribe?: TUnsubscribe;
+export const Aopp = () => {
+  const { t } = useTranslation();
 
-  public componentDidMount() {
-    this.setAccountCodeDefault();
-    this.unsubscribe = aoppAPI.subscribeAOPP(aopp => this.updateAOPP(aopp));
-    aoppAPI.getAOPP().then(aopp => this.setState({ aopp }));
-  }
+  const unsubscribe = useRef<TUnsubscribe>();
 
-  public componentWillUnmount() {
-    if (this.unsubscribe) {
-      this.unsubscribe();
+  const [accountCode, setAccountCode] = useState<accountAPI.AccountCode>('');
+  const [aopp, setAopp] = useState<aoppAPI.Aopp>();
+  const [shouldUpdateAccountCodeDefault, setShouldUpdateAccountCodeDefault] = useState(false);
+
+  const updateAOPP = (new_aopp: aoppAPI.Aopp) => {
+    if (new_aopp?.state === 'choosing-account'
+      && (
+        aopp?.state !== 'choosing-account'
+        || !equal(new_aopp.accounts, aopp?.accounts)
+      )) {
+      setShouldUpdateAccountCodeDefault(true);
     }
-  }
+    setAopp(new_aopp);
+  };
 
-  private updateAOPP(aopp: aoppAPI.Aopp) {
-    let shouldUpdateAccountCodeDefault = false;
-    this.setState(currentState => {
-      if (aopp?.state === 'choosing-account'
-        && (
-          currentState.aopp?.state !== 'choosing-account'
-          || !equal(aopp.accounts, currentState.aopp?.accounts)
-        )
-      ) {
-        shouldUpdateAccountCodeDefault = true;
-      }
-      return { aopp };
-    }, () => { // callback when state did update
-      if (shouldUpdateAccountCodeDefault) {
-        this.setAccountCodeDefault();
+  useEffect(() => {
+    setAccountCodeDefault(aopp, setAccountCode);
+    unsubscribe.current = aoppAPI.subscribeAOPP(aopp => updateAOPP(aopp));
```

Comment:
I think you don't need to store the unsubscribe callback in a ref.
```
const unsubscribe =  aoppAPI.subscribeAOPP...
....
return () => unsubscribe();
```

Reply:
oh true, no idea why I did this..